src/creature_instances.c

```diff
@@ -537,6 +548,133 @@ long process_creature_self_spell_casting(struct Thing* creatng)
     return 1;
 }
 
+/**
+ * @brief Check whether the given creature is suitable to cast ranged buff spell.
+ * This function is used for both combat and non-combat situations.
+ *
+ * @param creatng The creature being checked.
+ * @return CrInstance The instance index being set.
+ */
+CrInstance process_creature_ranged_buff_spell_casting(struct Thing* creatng)
+{
+    TRACE_THING(creatng);
+    SYNCDBG(8,"Processing %s(%d), act.st: %s, con.st: %s", thing_model_name(creatng), creatng->index,
+        creature_state_code_name(creatng->active_state), creature_state_code_name(creatng->continue_state));
+
+    if ((creatng->alloc_flags & TAlF_IsControlled) != 0)
+    {
+        // If this creature is under player's control.
+        return CrInst_NULL;
+    }
+
+    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+    if (cctrl->instance_id != CrInst_NULL) {
+        SYNCDBG(15, "This creature has an instance %s. Abort.", creature_instance_code_name(cctrl->instance_id));
+        return CrInst_NULL;
+    }
+
+    // Check validation of the caster.
+    if (creature_is_fleeing_combat(creatng) || creature_affected_by_spell(creatng, SplK_Chicken) ||
```

Comment:
This could probably be a sub-function. Something like 'creature_is_free_to_cast_ranged_buff`.

But is (or should) the function not only called when he is available for casting?

Reply:
I will move the code about validating caster to a sub-function.